test/GPv2Interaction.test.ts

```diff
@@ -0,0 +1,114 @@
+import { expect } from "chai";
+import { Contract } from "ethers";
+import { ethers, waffle } from "hardhat";
+
+describe("GPv2Interaction", () => {
+  const [deployer] = waffle.provider.getWallets();
+  let interactions: Contract;
+
+  beforeEach(async () => {
+    const GPv2Interaction = await ethers.getContractFactory(
+      "GPv2InteractionTestInterface",
+    );
+    interactions = await GPv2Interaction.deploy();
+  });
+
+  describe("execute", () => {
+    it("should pass on successful execution", async () => {
+      await expect(
+        interactions.executeTest({
+          target: ethers.constants.AddressZero,
+          callData: "0x",
+          value: 0,
+        }),
+      ).to.not.be.reverted;
+    });
+
+    it("should revert when interaction reverts", async () => {
+      const reverter = await waffle.deployMockContract(deployer, [
+        "function alwaysReverts()",
+      ]);
+      await reverter.mock.alwaysReverts.revertsWithReason("test error");
+
+      // TODO - update this error with concatenated version "GPv2 Interaction: test error"
+      await expect(
+        interactions.executeTest({
+          target: reverter.address,
+          value: 0,
+          callData: reverter.interface.encodeFunctionData("alwaysReverts"),
+        }),
+      ).to.be.revertedWith("test error");
+    });
+
+    it("should send Ether when value is specified", async () => {
```

Comment:
Maybe not in this PR, but it would be nice to se that this also works for EOAs and that if a contract has no payable fallback then the transaction reverts.

Reply:
I think these are nice tests worth having. I'll add them and we can back-port to `main` if we decide to not go with the `struct-calldata` branch.

Reply:
Done!